### PR TITLE
Vic 660 agency crud

### PR DESCRIPTION
**Pull request by adnanalicic**

### CODE REVIEW

File: src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java [hunk 1]
```diff
@@ -60,13 +64,34 @@ public ConsultantAdminResponseDTO findConsultant(String consultantId) {
...
+
+    if (containsNoValidField) {
+      sort.setField(FieldEnum.LASTNAME);
+      sort.setOrder(OrderEnum.ASC);
+    }
+  }
+
+  private Predicate<FieldEnum> providedSortFieldIgnoringCase(Sort sort) {
+    return field -> {
+      if (nonNull(sort.getField())) {
```
Comment (mebo4b) [inline src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java:87]: I think carinet is using the endpoint without any sort parameter. Could you also cover the case if sort is null?

File: src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java [hunk 2]
```diff
@@ -60,13 +64,34 @@ public ConsultantAdminResponseDTO findConsultant(String consultantId) {
...
+        return field.getValue().equalsIgnoreCase(sort.getField().getValue());
+      }
+      return false;
+    };
+  }
+
+
```
Comment (mebo4b) [inline src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java:94]: Please remove unneccessary line breaks

File: src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/ConsultantAgencyRelationCreatorService.java [hunk 3]
```diff
@@ -73,26 +73,28 @@ public void createNewConsultantAgency(String consultantId,
...
+
+    rocketChatAsyncHelper.addConsultantToSessions(consultant, agency, logMethod, TenantContext.getCurrentTenant());
 
     if (isTeamAgencyButNotTeamConsultant(agency, consultant)) {
       consultant.setTeamConsultant(true);
       consultantRepository.save(consultant);
     }
 
-    consultantAgencyService.saveConsultantAgency(buildConsultantAgency(consultant, agency.getId()));
+
```
Comment (mebo4b) [inline src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/ConsultantAgencyRelationCreatorService.java:97]: Could you please remove unneccesarry line breaks?

**Review by mebo4b — CHANGES_REQUESTED** [review]

File: src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java (hunk 1 above)
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java:90]: done.

File: src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java (hunk 2 above)
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java]: done

File: src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/ConsultantAgencyRelationCreatorService.java (hunk 3 above)
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/ConsultantAgencyRelationCreatorService.java]: done

File: src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java [hunk 4]
```diff
@@ -172,4 +205,70 @@ public AgencyConsultantResponseDTO findConsultantsForAgency(String agencyId) {
...
+   * @param newList      new list agencies that consultant belongs to
+   * @return filtered list of existing @{@link ConsultantAgency} ready for deletion
+   */
+  public List<Long> filterAgencyListForDeletion(String consultantId,
+      List<CreateConsultantAgencyDTO> newList) {
+    List<Long> newListIds = newList.stream().map(el -> el.getAgencyId())
+        .collect(Collectors.toList());
+    List<Long> persistedAgencyIds = consultantAgencyAdminService
+        .findConsultantAgencies(consultantId).getEmbedded().stream()
+        .map(el -> el.getEmbedded().getId()).collect(Collectors.toList());
```
Comment (mebo4b) [inline src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java:250]: Could you please find a bette rname for e1?

File: src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java [hunk 5]
```diff
@@ -172,4 +205,70 @@ public AgencyConsultantResponseDTO findConsultantsForAgency(String agencyId) {
...
+
+  /**
+   * Determines which from new agencies should be created.
+   *
+   * @param consultantId given consultant
+   * @param newList      new list of agencies that consultant belongs to
+   */
+  public void filterAgencyListForCreation(String consultantId,
+      List<CreateConsultantAgencyDTO> newList) {
+    List<Long> persistedAgencyIds = consultantAgencyAdminService
```
Comment (mebo4b) [inline src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java:264]: Please use var again and rethink the naming of e1

File: src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/ConsultantAgencyRelationCreatorService.java [hunk 6]
```diff
@@ -72,27 +73,41 @@ public void createNewConsultantAgency(String consultantId,
...
-    rocketChatAsyncHelper.addConsultantToSessions(consultant, agency, logMethod,
-        TenantContext.getCurrentTenant());
+    consultantAgencyService.saveConsultantAgency(buildConsultantAgency(consultant, agency.getId()));
+
+  }
+
+  public void completeConsultantAgencyAssigment(ConsultantAgencyCreationInput input,
+      Consumer<String> logMethod) {
+
+    Consultant consultant = this.retrieveConsultant(input.getConsultantId());
```
Comment (mebo4b) [inline src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/ConsultantAgencyRelationCreatorService.java:96]: Could you please use var again here?

**Review by mebo4b — APPROVED** [review]

File: src/main/java/de/caritas/cob/userservice/api/admin/controller/UserAdminController.java [hunk 7]
```diff
@@ -110,11 +110,11 @@ public ResponseEntity<Void> createConsultantAgency(@PathVariable String consulta
   @Override
   public ResponseEntity<Void> setConsultantAgencies(String consultantId,
       List<CreateConsultantAgencyDTO> agencyList) {
-    consultantAdminFacade.markConsultantAgenciesForDeletion(consultantId);
-    agencyList.forEach(agency ->
-        consultantAdminFacade.createNewConsultantAgency(consultantId, agency)
-    );
-
+    List<Long> agencyIdsForDeletions = consultantAdminFacade.filterAgencyListForDeletion(consultantId, agencyList);
```
[the remark this answers is not in the document]
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/admin/controller/UserAdminController.java:113]: i am old school java programmer. we did not have such fancy things back then ;)
but will change it.

File: src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java [hunk 8]
```diff
@@ -172,4 +205,70 @@ public AgencyConsultantResponseDTO findConsultantsForAgency(String agencyId) {
     var parsedAgencyId = Long.valueOf(agencyId);
     return this.consultantAgencyAdminService.findConsultantsForAgency(parsedAgencyId);
   }
+
+  /**
+   * Creates consultant agencies from given consultant id and agencies list set to status
+   * IN_PROGRESS.
+   *
+   * @param consultantId
```
[the remark this answers is not in the document]
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java:213]: done

File: src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java (hunk 4 above)
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java:250]: changed on all places

File: src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java (hunk 5 above)
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java:264]: done

File: src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/ConsultantAgencyRelationCreatorService.java (hunk 6 above)
Comment (adnanalicic) [reply src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/ConsultantAgencyRelationCreatorService.java:96]: done

**Pull request merged**